### ISSUE TITLE
minfi data load problem

### ISSUE THREAD

**Issue by chmcxz** [kasperdanielhansen/minfi#97]
I tried to run minfi under R3.3.3, and got error when using read.metharray.exp. Seemed that the RGset was generated, but both annotation and manifest were listed as "unknown", not as humanEPICmanifest and ilm10b2.hg19 even though I did install IlluminaHumanMethylationEPICmanifest and IlluminaHumanMethylationEPICanno.ilm10b2.hg19. therefore, when perform preprocessRaw, the error said "there is no package called 'Unknownmanifest'. One month ago, when I ran under R3.3.2, everything was perfectly fine. I also tried ChAMP which uses minfi to load data; same issue. What could be the problem? thanks.

**Comment by lorbaer** [comment]
Hi,
same problem here, however I'm still using R3.3.2
sessionInfo after loading minfi and hg19+manifest:
```
attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] IlluminaHumanMethylationEPICanno.ilm10b2.hg19_0.6.0 IlluminaHumanMethylationEPICmanifest_0.3.0         
 [3] minfi_1.20.2                                        bumphunter_1.14.0                                  
 [5] locfit_1.5-9.1                                      iterators_1.0.8                                    
 [7] foreach_1.4.3                                       Biostrings_2.42.1                                  
 [9] XVector_0.14.1                                      SummarizedExperiment_1.4.0                         
[11] GenomicRanges_1.26.4                                GenomeInfoDb_1.10.3                                
[13] IRanges_2.8.2                                       S4Vectors_0.12.2                                   
[15] Biobase_2.34.0                                      BiocGenerics_0.20.0                                

loaded via a namespace (and not attached):
 [1] mclust_5.2.3             base64_2.0               Rcpp_0.12.10             lattice_0.20-35          Rsamtools_1.26.1         digest_0.6.12           
 [7] R6_2.2.0                 plyr_1.8.4               RSQLite_1.1-2            httr_1.2.1               zlibbioc_1.20.0          GenomicFeatures_1.26.4  
[13] data.table_1.10.4        annotate_1.52.1          Matrix_1.2-8             preprocessCore_1.36.0    splines_3.3.2            BiocParallel_1.8.1      
[19] stringr_1.2.0            RCurl_1.95-4.8           biomaRt_2.30.0           rtracklayer_1.34.2       multtest_2.30.0          pkgmaker_0.22           
[25] openssl_0.9.6            GEOquery_2.40.0          quadprog_1.5-5           codetools_0.2-15         matrixStats_0.51.0       XML_3.98-1.6            
[31] reshape_0.8.6            GenomicAlignments_1.10.1 MASS_7.3-45              bitops_1.0-6             grid_3.3.2               nlme_3.1-131            
[37] xtable_1.8-2             registry_0.3             DBI_0.6-1                magrittr_1.5             stringi_1.1.3            genefilter_1.56.0       
[43] doRNG_1.6                limma_3.30.13            nor1mix_1.2-2            RColorBrewer_1.1-2       siggenes_1.48.0          tools_3.3.2             
[49] illuminaio_0.16.0        rngtools_1.2.4           survival_2.41-2          AnnotationDbi_1.36.2     beanplot_1.2             memoise_1.0.0   
```

**Comment by chmcxz** [comment]
have you tried: RGset@annotation=c(array='IlluminaHumanMethylationEPIC', annotation='ilm10b2.hg19')
after running RGset <- read.metharray.exp(targets=targets)? it worked for me. Good luck.

**Comment by kasperdanielhansen** [comment]
Could you tell me the output of
  nrow(RGset)
?

Best,
Kasper

**Comment by chmcxz** [comment]
> RGset <- read.metharray.exp(targets=targets)
> RGset@annotation=c(array='IlluminaHumanMethylationEPIC', annotation='ilm10b2.hg19')
> nrow(RGset)
Features 
 1051943 
> RGset
RGChannelSet (storageMode: lockedEnvironment)
assayData: 1051943 features, 56 samples 
  element names: Green, Red 
An object of class 'AnnotatedDataFrame'
  sampleNames: 201233640112_R01C01 201233640045_R02C01 ... 201233640045_R01C01 (56 total)
  varLabels: Sample_Name Sample_Well ... filenames (26 total)
  varMetadata: labelDescription
Annotation
  array: IlluminaHumanMethylationEPIC
  annotation: ilm10b2.hg19

**Comment by chmcxz** [comment]
Seems some cg are missing. there should be 863904 cg probes, and 2932 ch probes, but after preprocessRaw, there are only 863306 cg left. the number of ch probes are correct.

> raw=preprocessRaw(RGset)
Loading required package: IlluminaHumanMethylationEPICmanifest
> nrow(raw)
Features 
  866238 
> rawbeta=getBeta(raw)
> nrow(rawbeta)
[1] 866238
> rawCG=rawbeta[substr(row.names(rawbeta),1,2)=='cg',]
> nrow(rawCG)
[1] 863306

**Comment by lorbaer** [comment]
Hi,
```
> nrow(RGset)
Features 
 1051943 
> RGset
RGChannelSet (storageMode: lockedEnvironment)
assayData: 1051943 features, 2 samples 
  element names: Green, Red 
An object of class 'AnnotatedDataFrame': none
Annotation
  array: Unknown
  annotation: Unknown
```

> have you tried: RGset@annotation=c(array='IlluminaHumanMethylationEPIC', annotation='ilm10b2.hg19')
> after running RGset <- read.metharray.exp(targets=targets)? it worked for me. Good luck.

works like charm :) Thanks!

cheers

**Comment by kasperdanielhansen** [comment]
This issue is because you have scanned your arrays with a DMAP file which gives a certain number of probes (nrows in the RGChannelSet).  I have some code in minfi where I guess the array type based on the number of probes and I have tried to be specific.  I have expanded the guess in minfi 1.21.6 so if you re-read your data using that version, it should work.

It will be a few days before it appears on the build system.  Closing this issue
